src/sentry/static/sentry/app/views/acceptOrganizationInvite.tsx

```diff
@@ -0,0 +1,220 @@
+  needs2fa: boolean;
+  needsSso: boolean;
+  existingMember: boolean;
+  ssoProvider?: string;
+};
+
+type Props = AsyncViewProps;
+type State = AsyncViewState & {
+  inviteDetails: InviteDetails;
+  accepting: boolean | undefined;
+  acceptError: boolean | undefined;
+};
+
+class AcceptOrganizationInvite<P extends Props, S extends State> extends AsyncView<P, S> {
+  getEndpoints(): [string, string][] {
+    const {memberId, token} = this.props.params;
+    return [['inviteDetails', `/accept-invite/${memberId}/${token}/`]];
+  }
+
+  getTitle() {
+    return t('Accept Organization Invite');
+  }
+
+  makeNextUrl(path: string) {
+    return `${path}?${urlEncode({next: window.location.pathname})}`;
+  }
+
+  handleLogout = async (e: MouseEvent) => {
+    e.preventDefault();
+    await logout(this.api);
+    window.location.replace(this.makeNextUrl('/auth/login/'));
+  };
+
+  handleAcceptInvite = async () => {
+    const {memberId, token} = this.props.params;
+
+    this.setState({accepting: true});
+    try {
+      await this.api.requestPromise(`/accept-invite/${memberId}/${token}/`, {
+        method: 'POST',
+      });
+      browserHistory.push(`/${this.state.inviteDetails.orgSlug}/`);
+    } catch {
+      this.setState({acceptError: true});
+    }
+    this.setState({accepting: false});
+  };
+
+  get existingMemberAlert() {
+    const user = ConfigStore.get('user');
+
+    return (
+      <Alert type="warning" data-test-id="existing-member">
+        {tct(
+          'Your account ([email]) is already a member of this organization. [switchLink:switch accounts]?',
+          {
+            email: user.email,
+            switchLink: <Link href="#" onClick={this.handleLogout} />,
+          }
+        )}
+      </Alert>
+    );
+  }
+
+  get authenticationActions() {
+    const {inviteDetails} = this.state;
+
+    return (
+      <React.Fragment>
+        <p>
+          {t(
+            `To continue, you must either login to an existing Sentry account,
+             or create a new account.`
+          )}
+        </p>
+        {inviteDetails.needsSso && (
+          <p data-test-id="suggests-sso">
+            {tct(
+              `Note that [orgSlug] has enabled Single-Singn-On (SSO) using
+               [authProvider]. You may create an account by authenticating with
+               the organizations SSO provider.`,
+              {
+                orgSlug: inviteDetails.orgSlug,
+                authProvider: inviteDetails.ssoProvider,
+              }
+            )}
+          </p>
+        )}
+
+        <Actions>
+          {inviteDetails.needsSso ? (
+            <Button
+              data-test-id="sso-login"
+              priority="primary"
+              href={this.makeNextUrl(`/auth/login/${inviteDetails.orgSlug}/`)}
+            >
+              {t('Login with %s', inviteDetails.ssoProvider)}
+            </Button>
+          ) : (
+            <Button
+              data-test-id="create-account"
```

Comment:
I think for `<Button>` we should not use `data-test-id` -- instead use an `aria-label` selector, this gives us the additional benefit that our buttons are accessible.